Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,15 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def outdated?(greedy: false)
+      (greedy || !auto_updates && !version == :latest) &&
+      !version.nil? && (version.casecmp latest_installed_version) == 1
```

Comment:
> I have never seen a Cask get downgraded.

It has happened, but it’s mostly when the new version has a major bug that stops the app from running/installing, so even in those cases the downgrade can be considered an upgraded (you’d want the earlier version).

Either way, it has been discussed in the past we can’t trust any comparison other than “`version` is different” because our versions can be weird strings of characters. You can’t compare a hash to another and determine which was done later.